src/utils/ethereumUtils.js

```diff
@@ -515,7 +524,35 @@ async function parseEthereumUrl(data) {
   });
 }
 
+const calculateL1FeeOptimism = async (tx, provider) => {
+  try {
+    tx.value = toHex(tx.value);
+    tx.nonce = Number(await provider.getTransactionCount(tx.from));
+    delete tx.from;
+    tx.to = toChecksumAddress(tx.to);
+    tx.gasLimit =
+      toHex(tx.gasLimit) ||
+      toHex(tx.data === '0x' ? ethUnits.basic_tx : ethUnits.basic_transfer);
+    const currentGasPrice = store.getState().gas.selectedGasPrice?.value
+      ?.amount;
+    if (currentGasPrice) tx.gasPrice = toHex(currentGasPrice);
+
+    const serializedTx = serialize(tx);
+
+    const OVM_GasPriceOracle = new Contract(
+      OVM_GAS_PRICE_ORACLE,
+      optimismGasOracleAbi,
+      provider
+    );
+    const l1FeeInWei = await OVM_GasPriceOracle.getL1Fee(serializedTx);
+    return l1FeeInWei;
+  } catch (e) {
+    logger.log('error calculating l1 fee', e);
```

Comment:
no return value for when we fail to get the l1 fee

Reply:
that was intended.  If it's undefined it will default to null and won't be added to the L2 gas fee.